cpp/build-support/lint_cpp_cli.py

```diff
@@ -77,6 +77,7 @@ def lint_file(path):
 
 EXCLUSIONS = _paths('''\
     arrow/arrow-config.cmake
+    arrow/flight/sql/odbc/odbcabstraction/include/odbcabstraction/blocking_queue.h
     arrow/python/iterators.h
```

Comment:
Note to myself: After this PR is merged, I will make comment on https://github.com/apache/arrow/pull/40939:

 CLI/C++ support is being removed (https://github.com/apache/arrow/issues/45810) to allow usage of `mutex`, so adding the lint ignore here